src/constructs/lambda/lambda.ts

```diff
@@ -21,12 +22,28 @@ interface GuFunctionProps extends Omit<FunctionProps, "code"> {
   apis?: ApiProps[];
 }
 
+function defaultMemorySize(runtime: Runtime, memorySize?: number): number {
+  if (memorySize) {
+    return memorySize;
+  } else {
+    switch (runtime.family) {
+      case RuntimeFamily.JAVA:
+        return 1024;
```

Comment:
I based this on https://github.com/guardian/lambda-scala-boilerplate.g8/blob/master/src/main/g8/cfn.yaml#L67.